### PR TITLE
Add Constraints to device schema.

### PR DESCRIPTION
**Pull request by litghost**

### CODE REVIEW

File: interchange/DeviceResources.capnp [hunk 1]
```diff
@@ -382,4 +383,117 @@ struct Device {
     # Which cell have site local inverters?
     cellPins @1 : List(CellPinInversion);
   }
+
+  ######################################
+  # Placement constraints
+  #
+  # This section defines constrains required for valid placement above and
```
Comment (clavin-xlnx) [inline interchange/DeviceResources.capnp:390]: constrains --> constraints

File: interchange/DeviceResources.capnp [hunk 2]
```diff
@@ -382,4 +383,117 @@ struct Device {
...
+  # beyond routing constraints.
+  #
+  # This section has three sections:
+  #  - Tags
+  #  - Routed tags
+  #  - Cell constraints
+  #
+  # The tags sections defines a list of tags that take one of an exclusive set
+  # of states.  Tags are attached to an object with the FPGA fabric.
```
Comment (clavin-xlnx) [inline interchange/DeviceResources.capnp:399]: with --> within

File: interchange/DeviceResources.capnp [hunk 3]
```diff
@@ -382,4 +383,117 @@ struct Device {
...
+  # Currently tags can only be attached to either a site type or tile type.
+  # All instances of that site type or tile type will have it's own unique
+  # copy of the tag.  In order for the constraints to be considered "meet",
```
Comment (clavin-xlnx) [inline interchange/DeviceResources.capnp:402]: @litghost - Do you mean "met" here?

File: interchange/DeviceResources.capnp (hunk 2 above)
Comment (litghost) [reply interchange/DeviceResources.capnp:399]: Fixed

File: interchange/DeviceResources.capnp (hunk 1 above)
Comment (litghost) [reply interchange/DeviceResources.capnp:390]: Fixed

File: interchange/DeviceResources.capnp (hunk 3 above)
Comment (litghost) [reply interchange/DeviceResources.capnp:402]: Fixed

**Review by clavin-xlnx — COMMENTED** [review]
@litghost - constraints schema makes sense to me.  Seems like it should be robust in the scenarios that we have previously discussed.  I made some comments on lines 390, 399 and 402 of DeviceResources.capnp where there might be typos.  Otherwise, looks good to me.

**Review by clavin-xlnx — APPROVED** [review]

**Pull request merged**